### ISSUE TITLE
Have brew doctor check availability of Bintray CDN

### ISSUE THREAD

**Issue by wodow** [Homebrew/brew#10741]
**Provide a detailed description of the proposed feature**

Brew doctor could attempt a HTTPS request to the Bintray CDN as part of its sequence of checks.

**What is the motivation for the feature?**

Outages are not currently detected, as per https://github.com/Homebrew/brew/issues/10739#issue-818229396

**How will the feature be relevant to at least 90% of Homebrew users?**

Users caught within a CDN outage can stop further investigation if `brew doctor` alerts them to this.

**What alternatives to the feature have been considered?**

Manually check Twitter, Issues of this Github repo.

**Comment by amingilani** [comment]
```
$ curl https://homebrew.bintray.com/
curl: (35) error:1400410B:SSL routines:CONNECT_CR_SRVR_HELLO:wrong version number
```

**Comment by Rylan12** [comment]
I'm not sure what the benefit of having a check is. What would the `brew doctor` message say? This particular issue seems to be a one-off occurrence. Plus, we'll be moving away from Bintray soon, anyway. It's just going to tell users that things are broken (which should already be obvious). We can't provide a solution in that message, so it's not really anymore satisfying to users, right? `brew doctor` issues should be things that the user can do to fix their installation rather than a way to report issues that are out of their control.

**Comment by wodow** [comment]
> which should already be obvious

It wasn't obvious in this case. Anecdotally (Twitter etc), some users began trying to reinstall or otherwise mess with `openssl`, `curl`, and other stuff.

> We can't provide a solution in that message

"Please wait." -- i.e. this isn't a local fault; please do nothing -- would be a more useful solution than inspiring random hacking (for those so inclined).

Having said that, I understand the rationale to not try to add detection if the chance of outages is very rare.

**Comment by Bo98** [comment]
It makes sense, but the issue is we then need to have another remote server that reports whether there is another known outage or not (and such a status update would be done manually). As you say: it's impossible to 100% detect locally whether the issue was a global outage or a local fault.

There's something which is supposed to do that already: Bintray Status. Though on this occasion, that page remained showing "operational" during the outage.

**Comment by Rylan12** [comment]
One other thing I'd add (although other maintainers might disagree): I think it is helpful to receive (respectful and considerate) bug reports when something like this happens. Yesterday's incident happened on a Sunday where not nearly as many maintainers were active. Plus, it only affected a subset of the maintainers. Having users report the issues let us know how widespread the issue was so we could better judge what our reaction needed to be. In this case, we realized that it was widespread enough to warrant pushing a change and creating a new release (although Bintray fixed the issue before we actually made the release), whereas if it affected a smaller group of people we could have just recommended a smaller fix for just the affected group.

I do understand where you're coming from, though, as it's not immediately obvious what the problem is and whether it's an issue for just you or for everyone.
